pkg/utils/common.go

```diff
@@ -530,12 +541,14 @@ func UkiBootMode() state.Boot {
 }
 
 // SystemdBootConfReader reads a systemd-boot conf file and returns a map with the key/value pairs
-func SystemdBootConfReader(filePath string) (map[string]string, error) {
-	file, err := os.Open(filePath)
+func SystemdBootConfReader(fs v1.FS, filePath string) (map[string]string, error) {
```

Comment:
there's a version on the sdk now https://github.com/kairos-io/kairos-sdk/commit/f666b183705b7aee442c287ca07fd8cd469d38bc#diff-0df82f8689d3e8d881be0653ab4e9f8221c5835ec95a14930d11669e9c7f07efR358

Reply:
problem with this is, the sdk version doesnt have an abstrac FS to execute the operations on, so that makes it impossible to test it correctly without touching the real OS dir underneath (i.e. on test, create the proper files so we can test that its reading them and so on)

Which means we lose testing on this part if we use that implementation.

Would it be ok to maintain this for 3.0.0 and then try to move at least the v1.FS and its utils to the sdk and reuse it on the functions there? So everything can use that abstract FS which makes it easier to test?

@jimmykarily @mauromorales @mudler